app/models/job.rb

```diff
@@ -38,7 +38,7 @@ def started_by?(user)
   end
 
   def can_be_stopped_by?(user)
-    started_by?(user) || user.is_admin?
+    started_by?(user) || user.is_admin? || user.is_admin_for?(project)
```

Comment:
A Job can be stopped by the User that started it, a system-level admin or super-admin (covered by the is_admin? check), or by a project-level admin (covered by the is_admin_for? check).

I believe all the checks we're doing here are relevant.